proto/provenance/exchange/v1/market.proto

```diff
@@ -0,0 +1,143 @@
+  // fee_seller_settlement_ratios is the fee to charge a seller during settlement based on the price they are receiving.
+  // The price and fee denoms must be equal for each entry, and only one entry for any given denom is allowed.
+  repeated FeeRatio fee_seller_settlement_ratios = 6 [(gogoproto.nullable) = false];
+  // fee_buyer_settlement_flat is the flat fee charged to the buyer during settlement.
+  // Each coin entry is a separate option.
+  // When a bid is created, the settlement fees provided must contain one of these.
+  repeated cosmos.base.v1beta1.Coin fee_buyer_settlement_flat = 7 [(gogoproto.nullable) = false];
+  // fee_buyer_settlement_ratios is the fee to charge a buyer during settlement based on the price they are spending.
+  // The price and fee denoms do not have to equal. Multiple entries for any given price or fee denom are allowed, but
+  // each price denom to fee denom pair can only have one entry.
+  repeated FeeRatio fee_buyer_settlement_ratios = 8 [(gogoproto.nullable) = false];
+  // accepting_orders is whether this market is allowing orders to be created for it.
+  bool accepting_orders = 9;
+  // allow_user_settlement is whether this market allows users to initiate their own settlements.
+  // For example, the FillBids and FillAsks endpoints are available if and only if this is true.
+  // The MarketSettle endpoint is only available to market actors regardless of the value of this field.
+  bool allow_user_settlement = 10;
+  // access_grants is the list of addresses and permissions granted for this market.
+  repeated AccessGrant access_grants = 11 [(gogoproto.nullable) = false];
+  // req_attr_create_ask is a list of attributes required on an account for it to be allowed to create an ask order.
+  // An account must have all of these attributes in order to create an ask order in this market.
+  // If the list is empty, any account can create ask orders in this market.
+  //
+  // An entry that starts with "*." will match any attributes that end with the rest of it.
+  // E.g. "*.b.a" will match all of "c.b.a", "x.b.a", and "e.d.c.b.a"; but not "b.a", "xb.a", "b.x.a", or "c.b.a.x".
+  //
+  // An entry of exactly "*" will match any attribute, which is equivalent to leaving this list empty.
+  repeated string req_attr_create_ask = 12;
+  // req_attr_create_ask is a list of attributes required on an account for it to be allowed to create a bid order.
+  // An account must have all of these attributes in order to create a bid order in this market.
+  // If the list is empty, any account can create bid orders in this market.
+  //
+  // An entry that starts with "*." will match any attributes that end with the rest of it.
+  // E.g. "*.b.a" will match all of "c.b.a", "x.b.a", and "e.d.c.b.a"; but not "b.a", "xb.a", "c.b.x.a", or "c.b.a.x".
+  //
+  // An entry of exactly "*" will match any attribute, which is equivalent to leaving this list empty.
```

Comment:
Not a big deal, but why would we want to allow `*` and just not have an empty list as the option for `any`?  It would be consistent with Marker's required attributes then.

Reply:
I actually had the same thought, and it might simplify things if there is only one of these inputs.

Reply:
That thought occurred to me too. I didn't realize markers didn't allow simply `*` though. I'll take this functionality out of here.